Source/DafnyLanguageServer/DafnyLanguageServer.cs

```diff
@@ -19,21 +21,51 @@ public static class DafnyLanguageServer {
       }
     }
 
-    public static LanguageServerOptions WithDafnyLanguageServer(this LanguageServerOptions options, IConfiguration configuration) {
+    public static LanguageServerOptions WithDafnyLanguageServer(this LanguageServerOptions options,
+      IConfiguration configuration, CancellationTokenSource cancelLanguageServer) {
```

Comment:
I wasn't happy with this design but didn't know how to do it better, and was hoping you'd have an idea :D

Could you provide some code for the event alternative you mention? I'm not sure what you're suggesting.

Reply:
I've been checking omnisharp. Since it isn't using C#'s `event` feature I think we should leave that one aside since we don't need multiple registrations for now and follow omnisharp's `On*` convention with delegates. Therefore, I propose that we simply provide a callback for the `Kill` event. 

For example:
```cs    public delegate void OnDafnyLanguageServerKill();

    public static LanguageServerOptions WithDafnyLanguageServer(
      this LanguageServerOptions options,
      IConfiguration configuration,
      OnDafnyLanguageServerKill? onKill = null
    ) {
      return options
        .WithDafnyLanguage(configuration)
        .WithDafnyWorkspace(configuration)
        .WithDafnyHandlers()
        .OnInitialize((server, request, cancellationToken) => InitializeAsync(server, request, cancellationToken, onKill))
        .OnStarted(StartedAsync);
    }

    private static Task InitializeAsync(
      ILanguageServer server,
      InitializeParams request,
      CancellationToken cancellationToken,
      OnDafnyLanguageServerKill? onKill
    ) {
      void Kill() {
        logger.LogWarning("Shutting down language server because parent process died.");
        onKill?.Invoke();
      }
    }
```